torch/csrc/jit/tensorexpr/ir_deserializer.h

```diff
@@ -0,0 +1,234 @@
+namespace jit {
+namespace tensorexpr {
+
+using json = nlohmann::json;
+
+#define JSON_CACHE_GUARD()  \
+  if (cachedKernelObj(v)) { \
+    return;                 \
+  }
+
+/* Expression hasher providing comparable values representing sub-exprs.
+ * Uses memoization to avoid excessive recursion. */
+class TORCH_API IRDeserializer : public IRVisitor {
+ public:
+  bool cachedKernelObj(const json& json) {
+    size_t id = json["json_id"];
+    return JSONToObj_.count(id) != 0;
+  }
+
+  const Expr* getCachedExpr(const json& json) {
+    if (json.find("name") != json.end()) {
+      std::cout << " GETTING CACHED : " << json["name"];
+    }
+
+    auto it = JSONToObj_.find(json["json_id"]);
+    if (it != JSONToObj_.end()) {
+      const Expr* obj = dynamic_cast<const Expr*>(it->second);
+      assert(obj);
+      return obj;
+    }
+    assert(false);
+  }
+
+  const Stmt* getCachedStmt(const json& json) {
+    auto it = JSONToObj_.find(json["json_id"]);
+    if (it != JSONToObj_.end()) {
+      const Stmt* obj = dynamic_cast<const Stmt*>(it->second);
+      assert(obj);
+      return obj;
+    }
+    assert(false);
+  }
+
+  const Expr* dsExprImpl(const json& json);
+  const Expr* dsExpr(const json& json);
+  const std::vector<const Expr*> dsVecExpr(const json& json);
+
+  const Cast* deserializeCast(const json& json);
+  const Var* deserializeVar(const json& json);
+  const Ramp* deserializeRamp(const json& json);
+  const Broadcast* deserializeBroadcast(const json& json);
+  const IfThenElse* deserializeIfThenElse(const json& json);
+  // const Term * deserializeTerm(const json& json);
+  // const Polynomial * deserializePolynomial(const json& json);
+  const MaxTerm* deserializeMaxTerm(const json& json);
+  const MinTerm* deserializeMinTerm(const json& json);
+  const FunctionCall* deserializeFunctionCall(const json& json);
+  const Buf* deserializeBuf(const json& json);
+  const Intrinsics* deserializeIntrinsics(const json& json);
+
+  AtomicAdd* deserializeAtomicAdd(const json& json);
+  SyncThreads* deserializeSyncThreads(const json& json);
+  Let* deserializeLet(const json& json);
+  Store* deserializeStore(const json& json);
+  Block* deserializeBlock(const json& json);
+  For* deserializeFor(const json& json);
+  Free* deserializeFree(const json& json);
+  Cond* deserializeCond(const json& json);
+
+  Stmt* dsStmt(const json& v);
+  std::vector<Stmt*> dsVecStmt(const json& v);
+
+  const LongImm* deserializeLongImm(const json& json);
+  const ShortImm* deserializeShortImm(const json& json);
+  const BoolImm* deserializeBoolImm(const json& json);
+  const FloatImm* deserializeFloatImm(const json& json);
+  const DoubleImm* deserializeDoubleImm(const json& json);
+  const CharImm* deserializeCharImm(const json& json);
+  const HalfImm* deserializeHalfImm(const json& json);
+  const IntImm* deserializeIntImm(const json& json);
+  const ByteImm* deserializeByteImm(const json& json);
+
+  // void visit(const Load* v) override;
```

Comment:
You are missing an implementation for Load, which seems critical. This would also be a strong candidate to test.